openstack_plugin/resources/compute/flavor.py

```diff
@@ -63,6 +63,9 @@ def delete(openstack_resource):
     Delete flavor resource
     :param openstack_resource: Instance of openstack flavor resource.
     """
+    if not ctx.instance.runtime_properties.get(RESOURCE_ID):
+        ctx.logger.info('Flavor is already uninitialized.')
+        return
```

Comment:
@0lvin Instead of repeating this for every resource you can define this inside `with_openstack_resource` decorator as the following 
```
            try:
                # Prepare the openstack resource that need to execute the
                # current task operation
                resource = \
                    prepare_resource_instance(class_decl, ctx_node, kwargs)

                if use_external_resource(ctx_node, resource,
                                         existing_resource_handler,
                                         **existing_resource_kwargs):
                    return
                kwargs['openstack_resource'] = resource
               if not ctx.instance.runtime_properties.get(RESOURCE_ID):
                     ctx.logger.info('{0} is already uninitialized.'.format(resource.resource_type))
                     return
                func(**kwargs)
                update_runtime_properties_for_operation_task(operation_name,
                                                             ctx_node,
                                                             resource)
            except EXCEPTIONS as errors:
                _, _, tb = sys.exc_info()
                raise NonRecoverableError(
                    'Failure while trying to run operation:'
                    '{0}: {1}'.format(operation_name, errors.message),
                    causes=[exception_to_error_cause(errors, tb)])
``` 

CC @EarthmanT

Reply:
I will try to rework code to do logic in decorator.